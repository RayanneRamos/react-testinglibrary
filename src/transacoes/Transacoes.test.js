import React from 'react';
import { render } from '@testing-library/react';
import Transacoes from './Transacoes';

describe('Lista de transacoes', () => {
  it('O snapshot da lista de transações deve ser sempre o mesmo', () => {
    const listaDeTransacoes = [
      {
        id: 1,
        tipo: 'saque',
        valor: '20.00',
        data: '08/09/2020'
      },
      {
        id: 2,
        tipo: 'deposito',
        valor: '25.00',
        data: '08/10/2020',
      },
      {
        id: 3,
        tipo: 'saque',
        valor: '30.00',
        data: '10/09/2020',
      },
    ];
    const { container } =  render(<Transacoes transacoes={listaDeTransacoes} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});