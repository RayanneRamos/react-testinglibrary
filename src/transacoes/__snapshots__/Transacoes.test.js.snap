// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lista de transacoes O snapshot da lista de transações deve ser sempre o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      08/09/2020
    </p>
    <p />
    <p>
      R$ 
      20.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      08/10/2020
    </p>
    <p />
    <p>
      R$ 
      25.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10/09/2020
    </p>
    <p />
    <p>
      R$ 
      30.00
    </p>
  </div>
</div>
`;
